src/items/coherence.md

```diff
@@ -0,0 +1,79 @@
+## What is coherence and why do we care?
+
+Coherence means that for any given trait and type, there is one specific 
+implementation that applies. This is important for Rust to be easy to reason 
+about. When you write `<Foo as Bar>::trait_method`, the compiler needs to know
+what actual implementation to use.
+
+In languages without coherence, the compiler has to have some way to choose
+which implementation to use when multiple implementations could apply. Scala
+does this by having complex scope resolution rules for "implicit" parameters.
+Haskell (when a discouraged flag is enabled) does this by picking an impl
```

Comment:
This entire paragraph is irrelevant. Haskell (mainly) and other languages are only interesting for the purposes of "influences".